### PR TITLE
Add support for creation of a time_point_sec from an iso string

### PR DESCRIPTION
**Pull request by victorj8**
## Change Description

Add support for creation of a time_point_sec from an iso string #747  

## API Changes
- [ ] API Changes

## Documentation Additions
- [ ] Documentation Additions

### CODE REVIEW

File: libraries/eosiolib/core/eosio/time.hpp [hunk 1]
```diff
@@ -97,6 +100,25 @@ namespace eosio {
         static time_point_sec maximum() { return time_point_sec(0xffffffff); }
         static time_point_sec min() { return time_point_sec(0); }
 
+        static time_point_sec from_iso_string(std::string dateStr){
+           int y,mon,d,h,m;
+           float s;
+           sscanf(dateStr.c_str(), "%d-%d-%dT%d:%d:%fZ", &y, &mon, &d, &h, &m, &s);
```
Comment (swatanabe-b1) [inline libraries/eosiolib/core/eosio/time.hpp]: This is not strict enough.  Try strptime.  Also, you need to check the result, as parsing may fail.

File: tests/unit/time_tests.cpp [hunk 2]
```diff
@@ -364,6 +364,12 @@ EOSIO_TEST_BEGIN(time_point_sec_type_test)
    // friend bool operator>=(const time_point_sec&, const time_point_sec&)
    CHECK_EQUAL( (time_point_sec{1} >= time_point_sec{1}), true  )
    CHECK_EQUAL( (time_point_sec{1} >= time_point_sec{2}), false )
+
+   // --------------------------------------------------------------------
+   // static time_point_sec from_iso_string()
+   CHECK_EQUAL( time_point_sec::from_iso_string("1970-01-01T00:00:00.000").utc_seconds, 0);
+   CHECK_EQUAL( time_point_sec::from_iso_string("1998-06-15T08:13:12.000").utc_seconds, 897898392);
+   CHECK_EQUAL( time_point_sec::from_iso_string("2020-01-01T00:00:00.000").utc_seconds, 1577836800);
```
Comment (swatanabe-b1) [inline tests/unit/time_tests.cpp]: This doesn't match the sscanf format you used.

File: libraries/eosiolib/core/eosio/time.hpp (hunk 1 above)
Comment (victorj8) [reply libraries/eosiolib/core/eosio/time.hpp]: I added a check for this.

File: tests/unit/time_tests.cpp (hunk 2 above)
Comment (victorj8) [reply tests/unit/time_tests.cpp]: Fixed the format to match this.

File: libraries/eosiolib/core/eosio/time.hpp [hunk 3]
```diff
@@ -58,6 +61,24 @@ namespace eosio {
         const microseconds& time_since_epoch()const { return elapsed; }
         uint32_t            sec_since_epoch()const  { return uint32_t(elapsed.count() / 1000000); }
 
+        static time_point from_iso_string(std::string date_str) {
```
Comment (swatanabe-b1) [inline libraries/eosiolib/core/eosio/time.hpp]: Pass string by reference or use string_view.

**Comment by DenisCarriere** [comment]
@jeffreyssmith2nd and chance to roll this change into one of the latest EOSIO CDT releases?

**Pull request merged**